### PR TITLE
[Cleanup] Refactor some utility functions in Overlay tests

### LINKED ISSUE #3600: Refactor duplication in overlay tests
I noticed that we have quite a bit of duplication in our overlay tests, specifically functionality that accesses metrics from the application in order do some checks. Examples of functionality that is doing the same thing:
https://github.com/stellar/stellar-core/blob/ca2fb06059c15442cb4c9a8c89de1a8fc3579a39/src/overlay/test/FloodTests.cpp#L234

https://github.com/stellar/stellar-core/blob/ca2fb06059c15442cb4c9a8c89de1a8fc3579a39/src/overlay/test/FloodTests.cpp#L299

https://github.com/stellar/stellar-core/blob/ca2fb06059c15442cb4c9a8c89de1a8fc3579a39/src/overlay/test/OverlayTests.cpp#L1807

There might be other places as well. We should turn all these functions into free utility functions, and just reuse them across overlay tests.

### PR DESCRIPTION
**Pull request by hidenori-shinohara**
# Description

Resolves #3600 

The same utility functions in Overlay tests were defined in multiple locations. This PR puts them all in the same file so we don't have to continue to re-define them.

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [x] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/overlay/test/OverlayTestUtils.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+// Copyright 2022 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "main/Application.h"
```
Comment (marta-lokhova) [inline src/overlay/test/OverlayTestUtils.h:7]: No need for this include here - you can forward declare Application, and include it in the cpp file

File: src/overlay/test/OverlayTestUtils.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace stellar
+{
+
+std::uint64_t getOverlayFloodMessageCount(std::shared_ptr<Application> app,
```
Comment (marta-lokhova) [inline src/overlay/test/OverlayTestUtils.h:12]: use uint64_t for consistency

File: src/overlay/test/OverlayTestUtils.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "overlay/test/OverlayTestUtils.h"
+#include "medida/metrics_registry.h"
+#include "overlay/OverlayManager.h"
+#include "overlay/OverlayMetrics.h"
+
+namespace stellar
+{
+
```
Comment (marta-lokhova) [inline src/overlay/test/OverlayTestUtils.cpp:12]: should these functions be in their own namespace?

File: src/overlay/test/OverlayTestUtils.h (hunk 2 above)
Comment (marta-lokhova) [reply src/overlay/test/OverlayTestUtils.h:12]: also, `getOverlayFloodMessageCount` doesn't need to accessible outside of overlay test utils

**Review by marta-lokhova — CHANGES_REQUESTED** [review]

File: src/overlay/test/OverlayTestUtils.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include <memory>
+
+namespace stellar
+{
+
+class Application;
+
+namespace overlaytest
```
Comment (marta-lokhova) [inline src/overlay/test/OverlayTestUtils.h:14]: please use overlaytestutils to be consistent with the rest of the codebase

**Review by marta-lokhova — APPROVED** [review]
one minor rename request, otherwise LGTM. Squash the commits, and we'll merge this

File: src/overlay/test/OverlayTestUtils.cpp (hunk 3 above)
Comment (hidenori-shinohara) [reply src/overlay/test/OverlayTestUtils.cpp:13]: Done!

File: src/overlay/test/OverlayTestUtils.h (hunk 2 above)
Comment (hidenori-shinohara) [reply src/overlay/test/OverlayTestUtils.h]: Done

File: src/overlay/test/OverlayTestUtils.h (hunk 4 above)
Comment (hidenori-shinohara) [reply src/overlay/test/OverlayTestUtils.h]: Done!

**Review by nutty323 — APPROVED** [review]

**Review by nutty323 — APPROVED** [review]
4of6-ok

**Comment by marta-lokhova** [comment]
r+ fc50d5d26cc3c604de790ff7cb42d9173469e3df

**Pull request merged**